actions/utils.ts

```diff
@@ -359,3 +365,105 @@ export function isComposableCowCompatible(code: string): boolean {
+  } else {
+    try {
+      const selector = abi.slice(0, 10);
+      const message = ethers.utils.defaultAbiCoder.decode(
+        ["string"],
+        '0x' + abi.slice(10) // trim off the selector
+      )[0];
+      return { errorNameOrSelector: selector, message };
+    } catch {
+      // some weird parameter, just return and let the caller deal with it
+      return {};
+    }  
+  }
+}
+
+/**
+ * Parse custom reversion errors, irrespective of the RPC node's software
+ * 
+ * Background: `ComposableCoW` makes extensive use of `revert` to provide custom error messages. Unfortunately,
+ *             different RPC nodes handle these errors differently. For example, Nethermind returns a zero-bytes
+ *             `error.data` in all cases, and the error selector is buried in `error.error.error.data`. Other 
+ *             nodes return the error selector in `error.data`.
+ * 
+ *             In all cases, if the error selector contains a parameterised error message, the error message is
+ *             encoded in the `error.data` field. For example, `OrderNotValid` contains a parameterised error
+ *             message, and the error message is encoded in `error.data`.
+ * 
+ * Assumptions:
+ * - `error.data` exists for all tested RPC nodes, and parameterised / non-parameterised custom errors.
+ * - All calls to the smart contract if they revert, return a non-zero result at **EVM** level.
+ * - Nethermind, irrespective of the revert reason, returns a zero-bytes `error.data` due to odd message
+ *   padding on the RPC return value from Nethermind.
+ * Therefore:
+ * - Nethermind: `error.data` in a revert case is `0x` (empty string), with the revert reason buried in
+ *   `error.error.error.data`.
+ * - Other nodes: `error.data` in a revert case we expected the revert reason / custom error selector.
+ * @param error returned by ethers
+ */
+export const parseCustomError = (error: any): ParsedError => {
+  const { errorName, data } = error;
+
+  // In all cases, data must be defined. If it isn't, return early - bad assumptions.
+  if (!data) {
+    return {};
+  }
+
+  // If error.errorName is defined:
+  // - The node has formatted the error message in a way that ethers can parse
+  // - It's not a parameterised custom error - no message
+  // - We can return early
+  if (errorName) {
+    return { errorNameOrSelector: errorName };
+  }
+
+  // If error.data is not zero-bytes, then it's not a Nethermind node, assume it's a string parameterised
+  // custom error. Attempt to decode and return.
+  if (data !== "0x") {
+    console.log(data);
+    return rawErrorDecode(data)
+  } else {
+    // This is a Nethermind node, as `data` *must* be equal to `0x`, but we know we always revert with an
+    // message, so - we have to go digging ⛏️🙄
+    //
+    // Verify our assumption that `error.error.error.data` is defined and is a string.
+    // TODO: Is there a better way to do this?
+    if (typeof error?.error?.error?.data === "string") {
```

Comment:
nit: maybe save first `error?.error?.error?.data` into const, so you don't need to repeat in next line